src/Bridge/Doctrine/MongoDbOdm/Filter/DateFilter.php

```diff
@@ -0,0 +1,148 @@
+            $this->addMatch(
+                $aggregationBuilder,
+                $matchField,
+                self::PARAMETER_STRICTLY_BEFORE,
+                $values[self::PARAMETER_STRICTLY_BEFORE],
+                $nullManagement
+            );
+        }
+
+        if (isset($values[self::PARAMETER_AFTER])) {
+            $this->addMatch(
+                $aggregationBuilder,
+                $matchField,
+                self::PARAMETER_AFTER,
+                $values[self::PARAMETER_AFTER],
+                $nullManagement
+            );
+        }
+
+        if (isset($values[self::PARAMETER_STRICTLY_AFTER])) {
+            $this->addMatch(
+                $aggregationBuilder,
+                $matchField,
+                self::PARAMETER_STRICTLY_AFTER,
+                $values[self::PARAMETER_STRICTLY_AFTER],
+                $nullManagement
+            );
+        }
+    }
+
+    /**
+     * Adds the match stage according to the chosen null management.
+     */
+    private function addMatch(Builder $aggregationBuilder, string $field, string $operator, string $value, string $nullManagement = null)
+    {
+        try {
+            $value = new \DateTime($value);
+        } catch (\Exception $e) {
+            // Silently ignore this filter if it can not be transformed to a \DateTime
+            $this->logger->notice('Invalid filter ignored', [
+                'exception' => new InvalidArgumentException(sprintf('The field "%s" has a wrong date format. Use one accepted by the \DateTime constructor', $field)),
+            ]);
+
+            return;
+        }
+
+        $operatorValue = [
+            self::PARAMETER_BEFORE => '$lte',
+            self::PARAMETER_STRICTLY_BEFORE => '$lt',
+            self::PARAMETER_AFTER => '$gte',
+            self::PARAMETER_STRICTLY_AFTER => '$gt',
+        ];
+
+        if (null === $nullManagement || self::EXCLUDE_NULL === $nullManagement) {
+            $aggregationBuilder->match()->addAnd($aggregationBuilder->matchExpr()->field($field)->operator($operatorValue[$operator], $value));
+        } elseif (
+            (self::INCLUDE_NULL_BEFORE === $nullManagement && \in_array($operator, [self::PARAMETER_BEFORE, self::PARAMETER_STRICTLY_BEFORE], true)) ||
+            (self::INCLUDE_NULL_AFTER === $nullManagement && \in_array($operator, [self::PARAMETER_AFTER, self::PARAMETER_STRICTLY_AFTER], true)) ||
+            (self::INCLUDE_NULL_BEFORE_AND_AFTER === $nullManagement && \in_array($operator, [self::PARAMETER_AFTER, self::PARAMETER_STRICTLY_AFTER, self::PARAMETER_BEFORE, self::PARAMETER_STRICTLY_BEFORE], true))
+        ) {
+            $aggregationBuilder->match()->addOr(
+                $aggregationBuilder->matchExpr()->field($field)->operator($operatorValue[$operator], $value),
+                $aggregationBuilder->matchExpr()->field($field)->equals(null)
+            );
+        } else {
+            $aggregationBuilder->match()->addAnd(
```

Comment:
Multiple suggestions here:
* Fields are only evaluated if they are non-null. If a field is null, any of the comparison operators here won't match. The second expression is useless here.
* By default multiple conditions are always evaluated using `$and`, so you don't need to use `addAnd` in situations like these
* If you want to use multiple operators on the same field, call `field` once and chain the operator calls

With that in mind, this whole thing can be simplified to
```
$aggregationBuilder->match()->field($field)->operator($operatorValue[$operator], $value);
```

Reply:
I think I've done it like this because chaining a field with an operator in a match stage is not possible. You need to use `matchExpr`. But if I remove the line `$aggregationBuilder->matchExpr()->field($field)->notEqual(null)`, the else can be merged into the first `if`. We need Behat tests anyway for this part.